### ISSUE TITLE
Bring tagging to life

### ISSUE THREAD

**Issue by dtw-chriss** [BookStackApp/BookStack#868]
### For Feature Requests

I think the tagging is an great feature, but unless you know the tags used it is hard to search for them.   I'd me nice to have a tag page that somehow bring the tagging to life for the users - so they can quick see what is in the system.  Combine tagging with other attributes as recently updated/create or recent comments could be a pretty powerful view to better understand what is doing on in the application from a documentation standpoint.

**Comment by LazyLama** [comment]
Maybe you could add tagging to the search results. I have attached a mock up of what it could look like. I'm not a programmer, so I have no idea if this is even doable.
![image](https://user-images.githubusercontent.com/12070127/40917727-39d19e8c-67d2-11e8-88a0-bdd43039a95e.png)

**Comment by mendiromania** [comment]
The search system really needs a revamp. I'll probably do it if it's not done by next month.

**Comment by ssddanbrown** [comment]
Hi @dtw-chriss, Thanks for the suggestion. The title of this issue is fairly generic so it would be good if we could define some actionable points to achieve this.

Thanks @LazyLama for the mock-up. I think this is a good idea and is technically achievable. 

### Action Points _(Draft)_

* Include tags within search result listings as per @LazyLama 's mock-up above.
* Within the search interface, or perhaps on a sibling page, provide an overview of the tags that have been used which includes some view of tag popularity.

---

@dtw-chriss Does that sound like it would achieve what you're looking for? Let me know and I'll update the issue title and details.

@mendiromania It probably won't but you never know. I like to set a calm pace for this project :smile:. Be sure to [read the pull request](https://github.com/BookStackApp/BookStack#pull-requests) details before starting any work. I'd say a 'revamp' is large in scope.

**Comment by aljawaid** [comment]
This would be excellent. 

- please make the tags shown as clickable so they show all related articles when clicked.
- please consider screen space when using @LazyLama mockup.  Smaller tags under the title might be better?
- A separate curated page showing all the tags (clickable) sorte din alphabetical order would be an amazing live reference.  Maybe have a popup button on homepage for easier searching.  That would deffo bring the tags to life and encourage the user to search better.

**Comment by ssddanbrown** [comment]
Hi @tgrosinger,
After applying a tag, you can click on the label to search for tagged items with that label or you can click on the tag value (Right, lighter, side of it) to search on both the label and value.

Alternatively you can search using square brackets in the search bar, For example: `[Category=Guide]` will search for content with a tag label of `Category` that has a tag value of `Guide`. You can actually get quite complex with this and search using different operators such as `[Category!=Guide]` or `[Owners>5]`.

[More details here](https://www.bookstackapp.com/docs/user/searching/#advanced-search-syntax).

**Comment by tgrosinger** [comment]
@ssddanbrown that was very helpful, thank you!

**Comment by ssddanbrown** [comment]
Just coming back across this.

A tag overview was added as of the last release: https://www.bookstackapp.com/blog/bookstack-release-v21-11/
Upon this, tags were added to search results in an earlier release, and v21.11 enhanced this further with highlighting.

Since this covers both of the discussed action points above I'll close this off. Thanks all for you input 3 years ago :smile:
